### ISSUE TITLE
Update Bulkrax to v4

### ISSUE THREAD

**Issue by jeremyf** [scientist-softserv/britishlibrary#211]
At present the Browse Everything widget is broken in the v3.x branch of Bulkrax.  We want to upgrade to v4.4.0 (e.g. the latest version)

## Testing Criteria

- [x] Review upgrade notes to determine potential hazards of an upgrade.
  - In particular https://github.com/samvera-labs/bulkrax/pull/582
  - Jenny confirmed we are okay to proceed
- [x] Bulkrax importers works
- [x] Bulkrax importers section for cloud uploads is visible (e.g. I can indicate files to use; note there may not be files to use but the presence of the "Add Cloud File" is what matters most)

## Blocks

This upgrade (or back porting of functionality) blocks full functionality of:

- #173 
- #190 
- #172 

## Tasks

The [Upgrading a Gem within a Ruby Application](https://playbook-staging.notch8.com/e/en/dev/rails/upgrading-a-gem-within-a-ruby-application) playbook article outlines the general steps required for this work.

## Related commits:

- https://github.com/scientist-softserv/britishlibrary/pull/204
- https://github.com/samvera-labs/bulkrax/pull/619

The changes for samvera-labs/bulkrax#619 are part of the v4.3.0 and v4.4.0 branches (see below)

```
❯ cd ~/git/bulkrax ; git tag --contains 91f1b50
v4.3.0
v4.4.0
```

**Comment by jeremyf** [comment]
On 2022-12-08 I sent the [following via Slack](https://assaydepot.slack.com/archives/G0313NK2LJ0/p1670536415405869):

> One of the breaking aspects of BrowseEverything is in the Bulkrax section.  Currently BL is on v3.3.0 of Bulkrax (with a PR to move to v3.5.1).  However, for the fix to Bulkrax regarding the BrowseEverything bug, we need to do one of the following:
> 
> -   Upgrade to at least 4.3.x (latest being 4.4.0)
> -   Backport the fix from 4.3.0 to the 3.x branch
> 
> In reviewing the release notes of Bulkrax there is one breaking change: namely exporters now export in “pages” instead of one big file.  See the [breaking changes](https://github.com/samvera-labs/bulkrax/wiki/Upgrading-to-version-4.0).
> 
> My question to you is if this breaking change is a current change you don’t want to deal with?  If you’re okay with the breaking change, we can proceed with upgrading Bulkrax.  If not, we can look to backport the fix.  And if you have questions please ask.
> 
> My recommendation and preference is to upgrade Bulkrax.

**Comment by jeremyf** [comment]
[Jenny's response on 2022-12-09](https://assaydepot.slack.com/archives/G0313NK2LJ0/p1670585226249439?thread_ts=1670536415.405869&cid=G0313NK2LJ0):

> Thanks Jeremy, this is a great explanation. I am happy to go with your recommendation to upgrade Bx

**Comment by jeremyf** [comment]
I have moved this into deploy to staging because it's part of main and I've been testing local importers.

**Comment by grahamjevon** [comment]
Should bulkrax now be working in staging? I just tested a basic csv upload (metadata only) and the import failed in staging. The exact same csv file uploaded successfully in bl-demo.

**Comment by jeremyf** [comment]
Reading notes:

I QA-ed this update against the XML importer.  However, the imports that are failing are CSV.  In looking at the code; the assumptive `factory_class` is not being assigned for the record.  I'm delving deeper into this.

**Comment by jillpe** [comment]
We will be upgrading to Bulkrax 5 once this is complete

**Comment by grahamjevon** [comment]
**CSV importers stuck at pending**

While testing #280, I found that importers using the file_url template have become stuck at pending. After the first two importer attempts I noticed that the "current_he_institution" to "current_he_institution_name" column header change (#275) had only been reflected in the metadata only template in Google Sheets (and not the file_url and zip_folder templates). I therefore edited this column name before doing attempt 3. But this is also stuck at pending. Not sure if this is an issue with #275, #280, this ticket (#211), or something else.
